src/Chapter3.hs

```diff
@@ -753,6 +842,15 @@ parametrise data types in places where values can be of any general type.
   maybe-treasure ;)
 -}
 
+data Dragon p = Dragon p
+
+data TreasureChest x = TreasureChest
+    { treasureChestGold :: Int
+    , treasureChestLoot :: x
+    }
+
+data Lair p t = Lair (Dragon p) (Maybe (TreasureChest t))
```

Comment:
Great!
It is usually more comfortable to work with the record with named fields in projects, but this is a correct solution! 👍🏼